contracts/src/systems/hyperstructure/initialize_hyperstructure.cairo

```diff
@@ -0,0 +1,67 @@
+#[system]
+mod InitializeHyperStructure {
+    use eternum::alias::ID;
+    use eternum::components::hyperstructure::{HyperStructureResource, HyperStructure};
+    use eternum::components::position::{Position, Coord};
+
+    use dojo::world::Context;
+
+    use core::poseidon::poseidon_hash_span;
+    use core::traits::Into;
+    use core::array::{SpanTrait};
+
+    #[derive(Copy, Drop, Serde)]
+    struct ReqiuredResource {
+        resource_type: u8,
+        amount: usize
+    }
+
+    fn execute(ctx: Context, resources: Span<ReqiuredResource>, coord: Coord) -> ID {   
```

Comment:
the resource required should as a constant value somewhere, no passed in. We should just pass a 'hyperstructure_id' which will denote what the cost will be.

We will then need to track the balance of the structure built